### PR TITLE
Various fixes to enable compilation in Windows against Chocolatey's Gazebo binary

### PR DESCRIPTION
**Pull request by traversaro**
This permits to compile `gazebo-yarp-plugins` against the Gazebo binary version distributed by Microsoft in https://ms-iot.github.io/ROSOnWindows/ . 
 See the commit messages for the details of every fix. 
This PR includes a workaround for  https://github.com/ms-iot/ROSOnWindows/issues/48 .

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Authors: Enrico Mingo, Alessio Rocchi, Mirko Ferrati, Silvio Traversaro, Alessandro Settimi and Francesco Romano
 # CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
 
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.0)
```
[the remark this answers is not in the document]
Comment (traversaro) [reply CMakeLists.txt]: I agree, as this depends on YARP 3.1 that itself depends on CMake 3.5. The real reason is that 3.5 will involve several other cleanup, that I did not do at the moment. But you are right, I will update it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,19 +18,28 @@ include(GNUInstallDirs)
 
 # Finding dependencies
 find_package(YARP 3.0.101 REQUIRED)
-find_package(Boost REQUIRED serialization system)
-find_package(Protobuf REQUIRED)
```
[the remark this answers is not in the document]
Comment (traversaro) [reply CMakeLists.txt]: They are required, but `gazeboConfig.cmake` already calls them (ideally using `find_dependency)  so we can remove them.

[the remark this answers is not in the document]
Comment (diegoferigo) [reply CMakeLists.txt]: Got it!

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (traversaro) [reply CMakeLists.txt]: Done in https://github.com/robotology/gazebo-yarp-plugins/commit/80d749b55d30614472a746e677365efe9dfb7839 .

**Pull request merged**